sdk/highlight-ruby/highlight/test/highlight_test.rb

```diff
@@ -22,7 +22,7 @@ def test_record_log
 
   def test_trace
     Highlight::H.new('qe9y4yg1')
-    Highlight::H.instance.trace(nil, nil) do
+    Highlight::H.instance.trace(nil, nil, attrs: { "some.attribute" => 12 }) do
```

Comment:
What do you think about adding an assertion that these attributes are coming through? I might be misunderstanding the syntax here, but it looks like this invocation of `trace` is assuming `attrs` is defined as a keyword argument, but it [doesn't appear to be defined that way](https://github.com/highlight/highlight/pull/6338/files#diff-dc88d4e75408f5ac49bd6cc7c0dbbf54a2aef7faf81e7a1fc3ce9c0719575661R46).

Reply:
fixing that bug! what would be the best way to asset that? would i need to mock the opentelemetry sdk to see what arguments it is called with?

Reply:
Yeah, that's how I would expect to do it. All the testing I've done in the past was using RSpec, but you could use a syntax like:

```rb
expect_any_instance_of(OpenTelemetryTracer).to receive(:in_span).with('highlight-ctx', hash_including(
  "some.attribute": 12,
  "project_id": nil,
  ...
))`
```

Reply:
Also, no big deal if you don't want to add that right now. Just was thinking about how we could get a test failure to reproduce the issue if it wasn't being called correctly, but I think the test was already failing because of that.

Reply:
you're right, good idea to have a test for this. added this and i think it's passing locally?

Reply:
Nice! One minor thing... Since it's an expectation that something _will_ be called, I think you want to make the assertion before you invoke the method. Maybe not a problem if it's already working, but might be worth calling before calling any methods.

Reply:
Oh, also, I don't see `rspec` in our `Gemfile`. I think we are using minitest, and I don't think it has the `expect_any_instance_of` mocking API. I wonder if this is working 🤔

Reply:
oh :/ that makes a lot more sense. i was confused though because adding expect_any_instance_of wasn't complaining that it doesn't exist, but changing what i was asserting in existing didn't fail the test either. looking into how to mock with minitest and will make sure that it fails when i don't have the expected attributes passed

Reply:
after bashing my head at ruby for the past 2 hours, got a test that actually fails when i assert on incorrect attributes :) 
![image](https://github.com/highlight/highlight/assets/1351531/e8e155ce-192c-4936-b9f4-b88d6058d781)